code_examples/dapp/src/dapp/domainLinkageClaim.ts

```diff
@@ -1,19 +1,19 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import * as Kilt from '@kiltprotocol/sdk-js'
-
-let domainLinkageCtype: Kilt.ICType
-
-export function main() {
-  const did = 'did:kilt:4smcAoiTiCLaNrGhrAM4wZvt5cMKEGm8f3Cu9aFrpsh5EiNV'
-
+export type Parameter = {
+  domainLinkageCType: Kilt.ICType
+  didUri: Kilt.DidUri
+}
+export function main({ domainLinkageCType, didUri }: Parameter) {
   const claimContents: Kilt.IClaimContents = {
-    id: did,
-    origin: 'https://example-dApp.com'
+    id: didUri,
```

Comment:
that's the same as the claim owner though, right? do we need it?

Reply:
btw remember the VC discussion? Naming a claim `id` means we set ourselves up for disaster - `type` and `id` are reserved in that context

Reply:
These two are the required fields by the w3c spec. The domain linkage credential has `id` and `origin` fields.
For this PR we leave it as it is. This is only the improvement for the current strategy. Next step is to update this guide to the simple solution and in the future to have the spec complaint solution.

Reply:
I think there is a misunderstanding here; credentialSubject.id is not a claim, it's the credential subject's identifier, i.e. the claim owner. When we export KILT credentials to VCs, the credential owner becomes the credentialSubject.id. If we tried to export a claim with this CType we'd be adding the id twice.

Reply:
The [CType](https://github.com/KILTprotocol/ctype-index/blob/main/ctypes/0x9d271c790775ee831352291f01c5d04c7979713a5896dcf5e81708184cc5c643/ctype.json) was created with both `origin` and `id`. So the claim has to have both as well.

Reply:
And we fix this inconsistency afterwards? That would also work, not to break stuff right away.

Reply:
yeah i wouldn't handle that now